pkg/csi_driver/node.go

```diff
@@ -105,6 +105,21 @@ func (s *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublish
 
 	vc := req.GetVolumeContext()
 
+	// Check if the sidecar container was injected into the Pod
+	pod, err := s.k8sClients.GetPod(vc[VolumeContextKeyPodNamespace], vc[VolumeContextKeyPodName])
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "failed to get pod: %v", err)
+	}
+	gcsFuseSidecarImage := gcsFuseSidecarContainerImage(pod)
+	enableSidecarBucketAccessCheckForSidecarVersion := s.driver.config.EnableSidecarBucketAccessCheck && gcsFuseSidecarImage != "" && isSidecarVersionSupportedForGivenFeature(gcsFuseSidecarImage, SidecarBucketAccessCheckMinVersion)
+
+	if optInHostnetworkKSA && enableSidecarBucketAccessCheckForSidecarVersion {
+		// Sidecar bucket access check feature performs this vaildation in sidecar, for host network pods this doing a bucket access check in both node and sidecar will
```

Comment:
This comment has some typos change to: 

"Sidecar bucket access check feature performs this validation in sidecar. For host network pods, doing a bucket access check in both node and sidecar will lead to increased STS quota so we force skip the bucket access check on node. WI caches token for sidecar so it wouldn't result in quota increase."